### ISSUE TITLE
Wrong random formula for rewards

### ISSUE THREAD

**Issue by Pyrdacor** [Pyrdacor/Ambermoon#77]
The reward event is used to grant all kind of stuff like attribute or skill increasements, LP/SP refill, etc. A common use case are the wishing wells which grant attributes to the whole party.

Now there is a random flag for those events. If it is unset, you just get the given value. However if it is set, you should get a random value in the range 1 to the given value. At least this is how similar potions work and which is very logical.

The original code however just calculates a value in the range 0 to "given value minus 1". So in contrast to potion effects (which are implemented by special spells), there is not 1 added to the bonus value. This might lead to situations where a reward has no effect at all.

It could be on purpose but I highly doubt it. A reward should always be noticable and in general the value given as the reward value should be the max value you can get.

I adjusted it in Ambermoon Advanced and it is this way in the remake from the beginning. Still worth a fix in the original as well imho. Wishing wells will then grant 1 to 5 points for an attribute instead of 0 to 4. Potions already grant 1 to 5.

**Comment by a1exh** [comment]
I am not familiar with this in game but I do know that rewards in other games are often misused to cheat? 

Changing rewards in Ambermoon might have been implemented because someone found a way to cheat during testing and there was no time to take it out?

There is no way to apply a reward multiple times to gain an unfair advantage?

**Comment by Pyrdacor** [comment]
No. Unless you duplicate items like wishing coins though savegame manipulation.

I think the mentioned issue is not related to the things you mentioned. This is no "do nothing" reward. If you provide a value and don't use randomness, you will get exactly that number. For example 5 points in strength. The error only happens if random comes to play. Instead of a random value up to the given value, you will get a random value to the given value minus one and it can happen that you even get 0 points. Potions use almost the same formula but they add 1 in the end to have 1 to 5 instead of 0 to 4.

In pseudo code it is:

`int value = rand() mod 5`

And potions add 1 to this.
